ReactiveCocoaTests/Swift/FlattenSpec.swift

```diff
@@ -0,0 +1,200 @@
+//
+//  FlattenSpec.swift
+//  ReactiveCocoa
+//
+//  Created by Oleg Shnitko on 1/22/16.
+//  Copyright © 2016 GitHub. All rights reserved.
+//
+
+import Result
+import Nimble
+import Quick
+import ReactiveCocoa
+
+private extension SignalType {
+	typealias Pipe = (Signal<Value, Error>, Observer<Value, Error>)
+}
+
+private typealias Pipe = Signal<SignalProducer<Int, TestError>, TestError>.Pipe
+
+class FlattenSpec: QuickSpec {
+	override func spec() {
+
+		describe("Signal.switchToLatest") {
+
+			var pipe: Pipe!
+			var disposable: Disposable?
+
+			beforeEach {
+				pipe = Signal.pipe()
+				disposable = pipe.0.flatten(.Latest).observe { _ in }
+			}
+
+			afterEach {
+				disposable?.dispose()
+			}
+
+			context("disposing") {
+				var disposed = false
+
+				beforeEach {
+					disposed = false
+					pipe.1.sendNext(SignalProducer<Int, TestError> { _, disposable in
+						disposable += ActionDisposable {
+							disposed = true
+						}
+					})
+				}
+
+				it("should dispose inner signals when outer signal interrupted") {
+					pipe.1.sendInterrupted()
+					expect(disposed).to(beTrue())
+				}
+
+				it("should dispose inner signals when outer signal failed") {
+					pipe.1.sendFailed(TestError.Default)
+					expect(disposed).to(beTrue())
+				}
+
+				it("should not dispose inner signals when outer signal completed") {
+					pipe.1.sendCompleted()
+					expect(disposed).to(beFalse())
+				}
+			}
+		}
+
+		describe("Signal.merge") {
+
+			var pipe: Pipe!
+			var disposable: Disposable?
+
+			beforeEach {
+				pipe = Signal.pipe()
+				disposable = pipe.0.flatten(.Merge).observe { _ in }
+			}
+
+			afterEach {
+				disposable?.dispose()
+			}
+
+			context("disposing") {
```

Comment:
These tests are exactly the same, only difference being the `FlattenStrategy`.

Luckily `Quick` makes this really easy to solve!! You can abstract this entire `describe`:
```
func describeSignalFlattenDisposableTests(flattenStrategy: FlattenStrategy, name: String) {
    describe(name) {
        ...
    }
}
```
Then simply do:
```swift
describeSignalFlattenDisposableTests(.Latest, name: "Signal.switchToLatest")
describeSignalFlattenDisposableTests(.Merge, name: "Signal.merge")
describeSignalFlattenDisposableTests(.Concat, name: "Signal.concat")
```